cpp/include/cudf_test/nanoarrow_utils.hpp

```diff
@@ -416,3 +417,42 @@ get_decimal_precision()
   else
     return cudf::detail::max_precision<T>();
 }
+
+struct VectorOfArrays {
+  std::vector<nanoarrow::UniqueArray> arrays;
+  nanoarrow::UniqueSchema schema;
+  size_t index{0};
+
+  static int get_schema(ArrowArrayStream* stream, ArrowSchema* out_schema)
+  {
+    auto private_data = static_cast<VectorOfArrays*>(stream->private_data);
+
+    [[maybe_unused]] auto rc = ArrowSchemaDeepCopy(private_data->schema.get(), out_schema);
```

Comment:
I assume I missed a conversation :)
Why don't we check/return the error code here?

Reply:
Good find. This isn't actually new code, just moved from `from_arrow_stream_test.cpp`, but now is a good time to clean this up.